xtable-api/src/main/java/org/apache/xtable/conversion/ExternalCatalog.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.conversion;
+
+import java.util.Map;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import org.apache.xtable.model.catalog.CatalogType;
+
+/**
+ * This class represents the configuration for an external catalog. It holds information required to
+ * interact with the catalog, such as its identifier, type, table formats to sync and other catalog
+ * specific properties related to permissions, catalog uris, access-tokens etc.
+ */
+@Value
+@Builder
+public class ExternalCatalog {
+  /**
+   * An identifier to be used for the catalog if there are multiple catalogs of the same type but in
+   * different accounts or regions.
+   */
+  @NonNull String catalogIdentifier;
+
+  /** The type of catalog. */
+  @NonNull CatalogType catalogType;
+
+  /**
+   * The table formats that will be synced to this catalog along with their {@link TableIdentifier}.
+   * Eg: ICEBERG -> {marketing, price}, HUDI -> {marketing, price_hudi}, DELTA -> {delta_tables,
+   * price}
+   */
+  @NonNull Map<String, TableIdentifier> tableFormatsToSync;
```

Comment:
We have a concept of a `SourceTable` and `TargetTable` which both extend `ExternalTable`. Should we also have a base catalog class that we can use a source of information as well in the future?

Reply:
I don't see any fields in `ExternalCatalog` that can be added in the new class `SourceCatalog` right now, do you have any ?

Reply:
Catalog identifier and config would be needed to extract any metadata from source catalog right?

Reply:
Added classes for SourceCatalog and TargetCatalog.